src/drivers/dw/dma.c

```diff
@@ -639,13 +639,19 @@ static int dw_dma_stop(struct dma *dma, int channel)
 
 	spin_lock_irq(&dma->lock, flags);
 
-	ret = poll_for_register_delay(dma_base(dma) + DW_DMA_CHAN_EN,
-				      CHAN_MASK(channel), 0,
+	dw_write(dma, DW_CFG_LOW(channel), chan->cfg_lo | DW_CFG_CH_SUSPEND |
```

Comment:
Please add the comment and explanation here. Maybe include the doc reference here too.
I think the inline comments here is more important for else to understand this. This looks like magic to me

Reply:
Done.